### DISCUSSION TITLE
Gradients/Saving

### DISCUSSION THREAD

**Discussion by 100xCode** [TodePond/CellPond#287] [Help] [answered]
Hi there everyone. I was trying to build a life simulation that acted like a microscopic world. That was when I noticed three critical problems:

**1. Saving**
It appears the saving feature has not been implemented yet. I do know that saving was mentioned in #192, with TodePond even saying the system was fully operational. How do I utilize it? I don't want to lose my project! 😦

**2. Diamonds** <sub>(r.i.p)</sub>
The diamond shape was a very versatile shape when it was around. It allowed you to siphon RGB values from other squares into the rectangle slot of another square. It also allowed for addition and subtraction of said values. Combined with stamping, it was a powerful tool. Now that diamonds are gone, I can see that there's no way to get a red value and turn it into a green value for a different square.

**3. Græy**
gray + grey = græy.
The color "græy" is somewhat controversial. You can make a græy of any brightness by choosing the same red, green, and blue values for a certain square. The problem is, due to the way CellPond currently works, I can't select a gradient of græys. This means that the little white bacteria "rabbits" in my simulation can't reduce their græy value over time without losing the ability to move. I can't select a gradient of græys, so now a white-colored block whose red, green, and blue values have been decremented by one is no longer considered a rabbit. And I can't go with "rainbow", because that means that the movement paddle for my rabbits will also apply to the green plants I have. What can I do?

**Comment by TodePond** [comment, marked as answer]
1. Saving
To save, press Ctrl+S. Or save to the clipboard by pressing Ctrl+C.
To open, press Ctrl+O. Or load from the clipboard by pressing Ctrl+V. Or drag a file onto the page to load it.

2. Diamonds
Diamonds are gone ever since I changed the way that the virtual machine deals with fractal rules. You can't automate channel swapping currently. Will eventually replace this functionality with the triangle shape, but won't be for a while! If you really want, you can find an older version of cellpond from the repo history, but it won't have the newer improvements

3. Græy
Sorry it needs to be colourful otherwise the magic doesn't work. If you really really want a gradient of greys you can make 9 different greys, and group them together.

**Reply by 100xCode** [reply]
1. Thank you **SO** much! I was worried I would lose all my projects every time, but with a JSON viewer I can even edit the data. (get ready for shenanigans I'm going to break CellPond)
2. Oh, okay. Take your time, I know programming can be...well it's 90% debugging.
3. Oof. I could see this being solved if you had some way to channel the colors from the same square.

Thanks a lot! I've once again learned more about CellPond. I hope that one day I can open up CellPond with a tutorial to show me how it works, a finished controls menu, and much more! Best wishes!
